ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java

```diff
@@ -107,8 +108,8 @@ public class TransactionPool implements BlockAddedObserver {
   private final SaveRestoreManager saveRestoreManager = new SaveRestoreManager();
   private final Set<Address> localSenders = ConcurrentHashMap.newKeySet();
   private final EthScheduler.OrderedProcessor<BlockAddedEvent> blockAddedEventOrderedProcessor;
-  private final Map<VersionedHash, BlobsWithCommitments.BlobQuad> mapOfBlobsInTransactionPool =
-      new HashMap<>();
+  private final Map<VersionedHash, List<BlobsWithCommitments.BlobQuad>>
```

Comment:
This need to support concurrency, since notifications are sent async by potentially multiple threads

Reply:
Maybe switch to https://guava.dev/releases/3.0/api/docs/com/google/common/collect/Multimaps.html#synchronizedListMultimap(com.google.common.collect.ListMultimap) ?

Reply:
The access to the map was synchronized, but using a map implementation that does the synchronization for me is probably a better idea!